src/main/java/no/itera/bloggingplatform/repository/database/AbstractDbRepository.java

```diff
@@ -14,9 +14,21 @@ JdbcTemplate getJdbcTemplate() {
     return jdbcTemplate;
   }
 
-  Long getNextId() {
-    return getMaxId() + 1;
+  Long getNextId(String tableName) {
+    String sql = "SELECT max(id)+1 FROM " + tableName;
+    return jdbcTemplate.queryForObject(sql, Long.class);
   }
 
-  abstract Long getMaxId();
+  boolean exists(Long id, String tableName) {
+    String sql = "SELECT COUNT(id) FROM " + tableName + " WHERE id = ?";
```

Comment:
Toto nevyzera ako zly napad. Vo vseobecnosti je stale dobre vyextrahovat opakujuci sa kod. Len si treba potom dat bacha, aby sa do `tableName` premennej nedostal neziaduci kod (SQL Injection). V tychto podmienkach je to vsak ok.